### ISSUE TITLE
SVD run slow on big dataset / No progressbar in train method

### ISSUE THREAD

**Issue by baker-Xie** [NicolasHug/Surprise#122]
#### Description

1. The SVD algorithm run a little bit slow when in big dataset. (1M user and 400,000 items matrix)
    I trained for 20 hours and it didn't output a result. Can we use tensorflow to realize it based on surprise?
2. I can't not find the way to show the progressbar on algo.train() . When trainning in large dataset, one 
    may want a progressbar
This might not a issue, but I think the progressbar is very useful.

#### Steps/Code to Reproduce

df = pandas.read_table('trainset.txt',sep = '\t')
reader = Reader(rating_scale=(1, 5))
data = Dataset.load_from_df(df[['user_id', 'song_id', 'rating']], reader)
trainset = data.build_full_trainset()
algo = SVD(n_factors = 100,n_epochs = 1000,reg_all = 0.1)
algo.train(trainset)

#### Expected Results
The traing result
#### Actual Results
None
#### Versions

Linux-4.10.0-42-generic-x86_64-with-debian-stretch-sid
Python 3.6.2 |Anaconda custom (64-bit)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]

**Comment by NicolasHug** [comment]
1. You can perfectly try to implement SVD with a tensorflow backend on surprise yourself, though TBH I doubt it could be merged: I'm not a big fan of adding tons of dependencies and tensorflow is a huge one. I prefer to keep things simple, enven though it's not completely optimized. To make SVD faster,  a probably better option would be to use parrallelism. There are various papers on this, but I haven't read them in details. That being said, on your case, SVD is so slow because you're using 1000 epochs which is very high. on 5-fold CV, 20 epochs is enough to get a good RMSE. You can also try to reduce the number of factors, though it may hurt accuracy.
2. You can use the verbose parameter to SVD to print the current epoch.
